### PR TITLE
[#5277] improvement(cli): display index information for tables in the Gravitino CLI

### PR DESCRIPTION
**Pull request by orenccl**
### What changes were proposed in this pull request?

Add `--index` option to display index information on `Tables`.

Fix `TableDetails` command.

Update `cli/README.md` for table commands

### Why are the changes needed?

This change allows users to retrieve additional index information on `Tables`, providing more insights.

Fix: #5277

### Does this PR introduce _any_ user-facing change?

Yes, it adds the `--index` option to `CommandEntities.TABLE`.

### How was this patch tested?

1. Follow the instructions in the [cli README](https://github.com/apache/gravitino/tree/main/clients/cli) to build the CLI sub-project.
2. Start the Gravitino Playground.

To test, use a command like the following:
```
gcli -i table details --metalake metalake_demo --name catalog_mysql.db.iceberg_namespace_properties

gcli -i table details --metalake metalake_demo --name catalog_mysql.db.iceberg_namespace_properties --index

gcli -i table details --metalake metalake_demo --name catalog_postgres.hr.employee_performance
```

Check that the output matches the expected audit information.
image
image
image

### CODE REVIEW

File: clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java [hunk 1]
```diff
@@ -175,6 +176,10 @@ private void handleTableCommand() {
     if (CommandActions.DETAILS.equals(command)) {
       String table = name.getTableName();
       new TableDetails(url, ignore, metalake, catalog, schema, table).handle();
+
+      if (line.hasOption(GravitinoOptions.INDEX)) {
+        new ListIndexes(url, ignore, metalake, catalog, schema, table).handle();
+      }
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java:182]: This should be an if-else, only details or indexes should be displayed, not both.

**Comment by justinmclean** [comment]
Thanks for the PR. For it to be consistent with other commands, I would output in CSV format with an index (and its type) on each line.

File: clients/cli/README.md [hunk 2]
```diff
@@ -59,6 +60,75 @@ Before you can build and run this project, it is suggested you have the followin
...
+
+```bash
+gcli table delete --metalake <metalake_name> --name <catalog_name>.<schema_name>.<table_name>
+```
+
+Example:
+```bash
+gcli table delete --metalake metalake_demo --name catalog_mysql.db.iceberg_namespace_properties
+```
+
```
Comment (justinmclean) [inline clients/cli/README.md:131]: The README is for developers, not users, so the commands need to go in `docs/cli.md` not here.

**Comment by orenccl** [comment]
Done!

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListIndexes.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    StringBuilder all = new StringBuilder();
+    for (Index index : indexes) {
+      // Flatten the two-dimensional array fieldNames to a single dot-separated string
+      List<String> flattenedFieldNames = new ArrayList<>();
+      for (String[] fieldHierarchy : index.fieldNames()) {
+        // Join nested fields (e.g., "a.b.c") for each inner array
+        flattenedFieldNames.add(String.join(".", fieldHierarchy));
+      }
```
Comment (justinmclean) [inline clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListIndexes.java:74]: I think this code could be simpler; also, in most scenarios, the field names are single-level.

File: docs/cli.md [hunk 4]
```diff
@@ -353,6 +354,12 @@ gcli column list --metalake metalake_demo --name catalog_postgres.hr.departments
 gcli table details --metalake metalake_demo --name catalog_postgres.hr.departments --audit
 ```
 
+### Show table indexex
```
Comment (justinmclean) [inline docs/cli.md:357]: typo here

File: clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListIndexes.java (hunk 3 above)
Comment (orenccl) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListIndexes.java]: I've simplified the code, but I believe handling all cases, including nested structures, is still necessary. Does this look okay to you? #

Comment (justinmclean) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListIndexes.java]: @jerryshao What do you think?

Comment (jerryshao) [reply clients/cli/src/main/java/org/apache/gravitino/cli/commands/ListIndexes.java]: > I've simplified the code, but I believe handling all cases, including nested structures, is still necessary. Does this look okay to you?

I'm fine with the current implementation of using dot to join the field names.

**Comment by jerryshao** [comment]
My question from another PR (https://github.com/apache/gravitino/pull/5525#discussion_r1836037304), why do we need a new option to display the index information, can we just display all the table metadata information in one `details` command?

**Comment by justinmclean** [comment]
See https://github.com/apache/gravitino/pull/5525#discussion_r1837235935

**Comment by justinmclean** [comment]
@jerryshao can we move forward on this?

**Comment by jerryshao** [comment]
> @jerryshao can we move forward on this?

I've left the comment in another PR, I'm +0 on the current design choice, but I don't want to block things.

**Review by justinmclean — APPROVED** [review]
Thanks for your contribution

**Comment by justinmclean** [comment]
@jerryshao can you approve this? I approved this but I can't merge it.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**